go/vt/mysqlctl/replication.go

```diff
@@ -228,6 +228,27 @@ func (mysqld *Mysqld) GetServerUUID(ctx context.Context) (string, error) {
 	return conn.Conn.GetServerUUID()
 }
 
+// GetServerStatus returns the server statuses asked for.
+func (mysqld *Mysqld) GetServerStatus(ctx context.Context, statuses []string) ([]string, error) {
+	query := `SELECT variable_value FROM performance_schema.global_status WHERE variable_name IN (`
+	for _, status := range statuses {
```

Comment:
Previous comment is not correct about using `strings.Join` since we need to escape the variable name (though we do need the separator). The escaping can be done using `sqltypes.EncodeStringSQL()`.

Reply:
I would recommend something like this:
```
        // The format specifier is for any optional predicates.
        query := "SELECT variable_name, variable_value FROM performance_schema.global_status%s"
        where := strings.Builder{} // optional filters
        if len(names) > 0 {
		predicates.WriteString(" WHERE variable_name IN (")
        }
	for i, name := range names {
			if i > 0 {
				predicates.WriteByte(',')
			}
			predicates.WriteString(sqltypes.EncodeStringSQL(name))
		}
		predicates.WriteByte(')')
	}
        qr, err := mysqld.FetchSuperQuery(ctx, fmt.Sprintf(query, predicates.String())
```

Selecting the variable_name is necessary in order to return a map, which I think we really have to in order for this to be usable with multiple values.

I also think that the base query should be a const in the package (~ `getGlobalStatusQuery`).

Lastly, I also think that we should build a parsed query here rather than sending the raw unparsed one on. This will add some input validation to protect against intentional or not, unexpected/unwanted behaviors. For example, someone sending in something like `'); delete from mysql.user;` as a var name. For example:
```
	parsed := sqlparser.BuildParsedQuery(query, predicates.String())
        query, err := parsed.GenerateQuery(nil, nil)
        if err != nil {
             return nil, err
        }
        qr, err := mysqld.FetchSuperQuery(ctx, query)
```

Reply:
If we don't already, we should have a generic utility function that generates a comma separated string clause (as for `IN` clause) rather than implement it specifically in this function.

Reply:
We do. So if we do end up using `ParseAndBind`, which I think we should as noted above, we can leverage that to build the IN clause. Here's an example:

`... in %a ...`:
https://github.com/vitessio/vitess/blob/main/go/vt/vttablet/tabletmanager/vdiff/table_plan.go#L39

Building and binding the slice bind var:
https://github.com/vitessio/vitess/blob/377e1ddc99b9b1aeeae3eddb5a6ce575d1f5fc79/go/vt/vttablet/tabletmanager/vdiff/table_plan.go#L250-L262

Reply:
Fixed